src/main/java/gregtechfoodoption/recipe/chain/KebabChain.java

```diff
@@ -0,0 +1,154 @@
+                .input(dust,Salt,2)
+                .inputs(ONION_SLICE.getStackForm(8),Zest.getItemStack(2))
+                .fluidInputs(OliveOil.getFluid(1000))
+                .outputs(BargMeat.getItemStack(20))
+                .notConsumable(new IntCircuitIngredient(1))
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(27).duration(80)
+                .inputs(BargMeat.getItemStack(5),Zest.getItemStack(4),TOMATO_SLICE.getStackForm(4))
+                .output(KEBAB_BARG)
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(27).duration(80)
+                .inputs(BargMeat.getItemStack(5),Zest.getItemStack(4))
+                .fluidInputs(TomatoSauce.getFluid(100))
+                .output(KEBAB_BARG)
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(600).duration(200)
+                .inputs(KEBAB_BARG_COOKED.getStackForm(2),KEBAB_KUBIDEH_COOKED.getStackForm(),TOMATO.getStackForm(3),ONION.getStackForm(2),LEMON.getStackForm())
+                .fluidInputs(Stearin.getFluid(100),LemonExtract.getFluid(250))
+                .outputs(KEBAB_SOLTANI.getStackForm(2),SKEWER.getStackForm())
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(16).duration(100)
+                .inputs(ONION.getStackForm(3))
+                .input(dustTiny,Salt)
+                .inputs(SKEWER.getStackForm())
+                .outputs(KEBAB_ONION.getStackForm())
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(16).duration(100)
+                .inputs(TOMATO.getStackForm(2))
+                .input(dustTiny,Salt)
+                .inputs(SKEWER.getStackForm())
+                .outputs(KEBAB_TOMATO.getStackForm())
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(16).duration(400)
+                .inputs(CHUM.getStackForm(8),BANANA_PEEL.getStackForm(2),ONION.getStackForm(),MashedPotato.getItemStack(4),SKEWER.getStackForm(4))
+                .fluidInputs(Yolk.getFluid(200),Stearin.getFluid(400))
+                .outputs(KEBAB_CHUM.getStackForm(4))
+                .buildAndRegister();
+
+        CUISINE_ASSEMBLER_RECIPES.recipeBuilder().EUt(100).duration(1000)
+                .inputs(KEBAB_CHUM_COOKED.getStackForm(2))
+                .inputs(KEBAB_KUBIDEH_COOKED.getStackForm(5))
+                .inputs(BURGER_CHUM.getStackForm(14))
+                .input(Items.BUCKET)
+                .fluidInputs(RabbitStew.getFluid(250))
+                .fluidInputs(HotFryingOil.getFluid(250))
+                .fluidInputs(PotatoJuice.getFluid(4000))
+                .outputs(KEBAB_CHUM_BUCKET.getStackForm(),SKEWER.getStackForm(7))
+                .buildAndRegister();
+
+        GTFOUtils.addBakingOvenRecipes(KEBAB_KUBIDEH.getStackForm(),KEBAB_KUBIDEH_COOKED.getStackForm(),1000,800,4);
```

Comment:
Formatting can be improved by adding spaces after commas (you can use ctrl-shift-alt-L to do a whole file at once)